op-node/rollup/engine/engine_controller.go

```diff
@@ -105,7 +103,16 @@ type EngineController struct {
 	SyncDeriver SyncDeriver
 }
 
-func NewEngineController(engine ExecEngine, log log.Logger, metrics ECMetrics,
+// EngControllerInterface provides direct calls into the EngineController that
+// external components can use instead of emitting events.
+type EngControllerInterface interface {
```

Comment:
nit: what do you think about moving this to `attributes/attributes.go` and renaming it to `EngineController`?